app/src/main/java/org/simple/clinic/main/TheActivityController.kt

```diff
@@ -124,20 +108,20 @@ class TheActivityController @Inject constructor(
   fun initialScreenKey(): FullScreenKey {
     val localUser = userSession.loggedInUser().blockingFirst().toNullable()
 
+    val userDisapproved = localUser?.status == UserStatus.DisapprovedForSyncing
+
     val canMoveToHomeScreen = when (localUser?.loggedInStatus) {
       NOT_LOGGED_IN, RESETTING_PIN, UNAUTHORIZED -> false
       LOGGED_IN, OTP_REQUESTED, RESET_PIN_REQUESTED -> true
       null -> false
     }
 
     return when {
-      canMoveToHomeScreen -> HomeScreenKey()
+      canMoveToHomeScreen && !userDisapproved -> HomeScreenKey()
+      userDisapproved -> AccessDeniedScreenKey(localUser?.fullName!!)
```

Comment:
Do you feel that it's worth adding a test for this as well?

Reply:
Yes. Done.